horovod/_keras/__init__.py

```diff
@@ -84,6 +84,9 @@ def variables(self):
                 return super(self.__class__, self).variables()
             return self.get_weights()
 
+        if version.parse(tf.__version__) >= version.parse('2.12.0'):
```

Comment:
Thanks @maxhgerlach I could figure out this one.
So TF changed the attribute from method to property? :)

Reply:
They did, yes: https://github.com/keras-team/keras/commit/50e0b18ceeb370157f36a387115c5ae1a16fca47

There's a little compatibility shim so code using an optimizer still works with both `.variables` and with `.variables()`.

Cheers!